### ISSUE TITLE
Assorted review comments from CWG-5

### ISSUE THREAD

**Issue by tkoeppe** [cplusplus/draft#4403]
See #4379 for context.

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529197110

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r528984720

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r528987386

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529003267

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r528987146

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529018079

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529026436

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529028230

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r530176892

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r530681396

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r530874564

Please consider opening a separate issue or a PR for this.

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r530894725

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r530907561

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r531133933

PR welcome.

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r531163099

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r531192830

Might be worth a small PR?

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r532044227 (request for example)

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r532045961 -- typo, @opensdh?

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r533045446 (define "component name")

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r533048853 and https://github.com/cplusplus/draft/pull/4379#discussion_r533050251 -- meaning of parentheticals

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r533055065 - @jensmaurer?

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r533059181 -- prefer to use bullets

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r531348083 -- request for rewording

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r531230199

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r531098787 (@tkoeppe )

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r530775310

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r530208739 -- request for rewording

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r531169215

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r531205309 -- ambiguous pronoun

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r531210196 -- placement of cross reference

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r531212307, https://github.com/cplusplus/draft/pull/4379#discussion_r531216483 -- questions for clarification

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529032752 -- define "program point"

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529036870 -- define "type-only"

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529063125 - request for rewording

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529092079 -- what's the point?

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529099209 -- what is it?

**Comment by opensdh** [comment]
Having written some things at #4379, I now realize this is the correct place for practical reasons; I'll make further comments here.  I don't have the permissions to edit the points in place, though, so I'm not sure how to attach them to the correct comment-identifying comment.

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529104829

**Comment by tkoeppe** [comment]
@opensdh: Thank you! I copied some of the material out. I don't yet know how to organize this best. Could you maybe just start a new post, leading with the original comment link, and maybe (if you can) also link the comment here that lists it? But I can also do that latter step later.

I'm just trying to somehow keep track of everything, this is far from thought out :-(

Thanks again!

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529110651

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529121667 -- "constructor name"

**Comment by tkoeppe** [comment]
@opensdh: Dealing with the parentheticals is perhaps the most fruitful post-merge cleanup for now, since it seems to have the biggest potential for confusion.

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529127228 -- request for rewrite

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529194123 -- clarify pronoun

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529188884

**Comment by tkoeppe** [comment]
https://github.com/cplusplus/draft/pull/4379#discussion_r529189790 -- "result of what"

**Comment by opensdh** [comment]
[subject](#issuecomment-738134341)

We need to actually say that such a scope is introduced, which was the motivation for the current phrasing.  If definitions must be in terms of "to be", we can say

> A parameter-declaration-clause _P_ introduces a scope.  A _function parameter scope_ is any such scope.  The scope includes _P_.

(where the last bit is put in its own sentence to connect to the subsequent potential expansions of the scope).  If we can use "called", we can use the simpler formulation

> A parameter-declaration-clause _P_ introduces a scope, called a _function parameter scope_, that includes _P_.

Whatever the rules for definitions are, we should [document them](/cplusplus/draft/wiki/Specification-Style-Guidelines).

**Comment by opensdh** [comment]
P1787R6 doesn't introduce this [(poor) definition of "extends"](#issuecomment-738136712); that should be a separate issue.

**Comment by opensdh** [comment]
[Proposal](#issuecomment-738139008):

> or that nominates a class other than with an *elaborated-type-specifier* of the form `class` *identifier*

**Comment by opensdh** [comment]
I think using ["verify"](#issuecomment-738333027) would be wrong: the name that is dependent is looked up _before_ it is known to be dependent, not as a double-check.  (That's not to say we can't find a better word!)

**Comment by opensdh** [comment]
The ["X or Y and Z"](#issuecomment-743287243) cannot be reasonably interpreted as "(X or Y) and Z" because Z refers to an object identified by Y.  We can of course use bullets to avoid the reader having to figure this out.

**Comment by tkoeppe** [comment]
It's perferctly possible that I eventually messed up the copy paste and copied something twice!

**Comment by opensdh** [comment]
The ["therefrom"](#issuecomment-743304150) is "from [that] explicit instantiation of that entity" (again).

**Comment by opensdh** [comment]
Either of ["whose" -> "which" or "in whose" -> "whose"](#issuecomment-743304253) would change the meaning of the sentence, as would rewording to omit "search" (which causes the naming class to be the derived class, not the base class, unless `d.B::x` is used).

**Comment by opensdh** [comment]
Jens' [interpretation](#issuecomment-743305630) is correct; the reachability check is between D3 and D2, although I haven't tried to construct an example where this differs from checking D1/D2.

**Comment by opensdh** [comment]
The ["member of a class that is the current instantiation"](#issuecomment-743308572) is preexisting.  A lookup context is an entity; the check against "the current instantiation" is complicated by the existing confusion between *id-expression*s and entities in the use of that phrase.

**Comment by opensdh** [comment]
The phrase ["program point"](#issuecomment-743310967) is preexisting (although often just as the word "point").

**Comment by opensdh** [comment]
The ["point of the lookup"](#issuecomment-743312405) is the program point from which the lookup is performed.  All lookup has such a point, although in some cases one lookup is described in terms of another which is supposed to take place at the same point without repeating "from a point _P_" on both sides.

**Comment by opensdh** [comment]
Here, ["it"](#issuecomment-743312871) refers to the "class, namespace, or enumeration" in question.

**Comment by opensdh** [comment]
The ["(original) lookup context"](#issuecomment-743313119) is the lookup context of _Q_, not the one used for the surrogate lookups in bullets 2 and 3.

**Comment by opensdh** [comment]
["Such a constructor name"](#issuecomment-743314321) is preexisting.

**Comment by opensdh** [comment]
How can ["its"](#issuecomment-743316244) (be construed to) refer to a plural subject?

**Comment by opensdh** [comment]
["The result is the entity"](#issuecomment-743316571) is preexisting (the diff is misleading here).

**Comment by tkoeppe** [comment]
Thanks a lot for all the clarifications!
